R/mediationanalysis.R

```diff
@@ -340,26 +341,44 @@ MediationAnalysis <- function(jaspResults, dataset, options, ...) {
   dirtab[["ci.lower"]] <- pe_dir$ci.lower
   dirtab[["ci.upper"]] <- pe_dir$ci.upper
   dirtab$addFootnote(foot_message)
+
+  # Fill indirect effects
   pe_ind <- pe[pe$op == ":=" & vapply(gregexpr("_", pe$lhs), length, 1) == 3, ]
-  
-  indtab[["x"]]        <- rep(options$predictor, each = length(options$mediators) * length(options$dependent))
+  # get predictors, mediators, outcome combinations
+  terms <- strsplit(pe_ind[["label"]], "_")
+  termsCombinations <- list(
+    x = as.integer(gsub("x", "", sapply(terms, "[[", 2))),
+    m = as.integer(gsub("m", "", sapply(terms, "[[", 3))),
+    y = as.integer(gsub("y", "", sapply(terms, "[[", 4)))
+  )
+
+  indtab[["x"]]        <- .unv(options[["predictor"]][termsCombinations[["x"]]])
   indtab[["op1"]]      <- rep("\u2192", nrow(pe_ind))
-  indtab[["m"]]        <- rep(options$mediators, length(options$dependent) * length(options$predictor))
+  indtab[["m"]]        <- .unv(options[["mediators"]][termsCombinations[["m"]]])
   indtab[["op2"]]      <- rep("\u2192", nrow(pe_ind))
-  indtab[["y"]]        <- rep(rep(options$dependent, each = length(options$mediators)), length(options$predictor))
+  indtab[["y"]]        <- .unv(options[["dependent"]][termsCombinations[["y"]]])
   indtab[["est"]]      <- pe_ind$est
   indtab[["se"]]       <- pe_ind$se
   indtab[["z"]]        <- pe_ind$z
   indtab[["pvalue"]]   <- pe_ind$pvalue
   indtab[["ci.lower"]] <- pe_ind$ci.lower
   indtab[["ci.upper"]] <- pe_ind$ci.upper
   indtab$addFootnote(foot_message)
-  
+
+
+  # Fill total effects
   pe_tot <- pe[pe$op == ":=" & substr(pe$lhs, 1, 3) == "tot",]
-  
-  tottab[["lhs"]]      <- rep(options$predictor, length(options$dependent))
+
+  # get predictors, outcome combinations
+  terms <- strsplit(pe_tot[["label"]], "_")
+  termsCombinations <- list(
+    x = as.integer(gsub("x", "", sapply(terms, "[[", 2))),
+    y = as.integer(gsub("y", "", sapply(terms, "[[", 3)))
```

Comment:
Could you use `vapply` here instead of `sapply`? Same on lines 350-352 and 422-423.